### PR TITLE
Bluetooth: Controller: Fix multiple Central role spacing

### PR DESCRIPTION
**Pull request by cvinayak**
Find and match only Central role ticker when applying the
BT_CTLR_CENTRAL_SPACING value.

Use the EVENT_TICKER_RES_MARGIN_US value as +/- jitter
margin when ticker advanced scheduling is used to group
radio events.

When using CONFIG_BT_CTLR_CENTRAL_SPACING, do not add the
+/- jitter to the ticks_slot.

Move the use of rx chain delay value used to calculate the
window offset into lll_scan_prepare_connect_req() instead of
current two places, one for legacy and other for extended
connection setup.

Update Advanced Scheduling implementation to use remainder
value in precisely calculating the connection request window
offset in microseconds.

Fix window offset calculation which is missing the radio
rx ready delay timing be subtracted because the connection
offset from the start of the radio for scan window to the
end of the ADV_IND PDU includes the radio rx ready delay
timing, where as the determined window offset by advanced
scheduling calculates the offset from the ticks_at_expire
of the scan window to an active central connection's
ticks_at_expire.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_sched.c [hunk 1]
```diff
@@ -170,33 +175,61 @@ int ull_sched_adv_aux_sync_free_slot_get(uint8_t user_id,
...
+		 * Hence, use CONFIG_BT_CTLR_CENTRAL_SPACING without margin.
+		 *
+		 * If actual ticks_slot in microseconds is larger than
+		 * CONFIG_BT_CTLR_CENTRAL_SPACING, then add margin between
+		 * central radio events.
+		 */
+		slot_us = HAL_TICKER_TICKS_TO_US(ticks_slot);
+		if (slot_us > CONFIG_BT_CTLR_CENTRAL_SPACING) {
+			central_spacing_us = slot_us;
+			central_spacing_us += (EVENT_TICKER_RES_MARGIN_US << 1);
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_sched.c:209]: Many places we simply add `EVENT_TICKER_RES_MARGIN_US` and some places we add `EVENT_TICKER_RES_MARGIN_US << 1`. Can you explain why/when we do one or the other?

File: subsys/bluetooth/controller/ll_sw/ull_sched.c [hunk 2]
```diff
@@ -330,7 +376,7 @@ static uint8_t after_match_slot_get(uint8_t user_id, uint32_t ticks_slot_abs,
 	 * event, and an ~30.517 us before next event. Hence 8 time of ceil
 	 * value 16 us (30.517 / 2).
 	 */
-	ticks_slot_abs += HAL_TICKER_US_TO_TICKS(EVENT_JITTER_US << 3);
+	ticks_slot_abs += HAL_TICKER_US_TO_TICKS(EVENT_TICKER_RES_MARGIN_US << 2);
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_sched.c:379]: Do the above comment need to be updated?

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan.c [hunk 3]
```diff
@@ -287,7 +289,9 @@ void lll_scan_prepare_connect_req(struct lll_scan *lll, struct pdu_adv *pdu_tx,
 		*conn_space_us = conn_offset_us;
 		pdu_tx->connect_ind.win_offset = sys_cpu_to_le16(0);
 	} else {
-		uint32_t win_offset_us = lll->conn_win_offset_us;
+		uint32_t win_offset_us =
+			lll->conn_win_offset_us +
+			radio_rx_ready_delay_get(phy, PHY_FLAGS_S8);
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan.c:294]: no. Window Offsets are max 16-bit of 1.25 ms units would max at 8.2 seconds and does not overflow 32-bit when stored in microseconds.

File: subsys/bluetooth/controller/ll_sw/ull_sched.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_sched.c:209]: The margin between two radio events that jitter is `EVENT_TICKER_RES_MARGIN_US << 1` and when inserting a new radio event between two existing event then the margin used while finding a free space is `EVENT_TICKER_RES_MARGIN_US << 2`.

There are only 3 occurrences of use of `EVENT_TICKER_RES_MARGIN_US` in ull_sched.c of above two types.

**Review by Thalley — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**